src/Core/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -382,9 +375,24 @@ private void LogPrimaryKeys()
         public abstract Type SqlToCLRType(string sqlType);
 
         /// <summary>
-        /// Takes a string version of a sql data type and returns its corresponding DbType.
+        /// Updates a table's SourceDefinition object's metadata with whether any enabled insert/update DML triggers exist for the table.
+        /// This method is only called for tables in MsSql.
+        /// </summary>
+        /// <param name="entityName">Name of the entity.</param>
+        /// <param name="schemaName">Name of the schema in which the table is present.</param>
+        /// <param name="tableName">Name of the table.</param>
+        /// <param name="sourceDefinition">Table definition to update.</param>
+        public virtual Task PopulateTriggerMetadataForTable(string entityName, string schemaName, string tableName, SourceDefinition sourceDefinition)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Takes a string version of a sql date/time type and returns its corresponding DbType.
+        /// Returns null when the provided sql type is not a date/time type.
         /// </summary>
-        public abstract DbType DatetimeSqlTypeToDbype(string sqlType);
+        /// <param name="sqlDbTypeName">Name of the SqlDb type.</param>
+        public virtual DbType? SqlDbTypeToDbType(string sqlDbTypeName) => null;
```

Comment:
I see it's usage is specific to mssql. Perhaps a more generic name then because it doesn't really make sense to have this in pgsql and mysql. I only say that because `SqlDbType` is specific to mssql.

Maybe `ResolveDbType(string s)` then when it comes to PG and MySQL we can find the mapping between their specific types and the `DbType` enum.

Reply:
> because SqlDbType is specific to mssql.

Why does PgSql, MySql not qualify for `sql`?

Reply:
`SqlDbType` refers to SQL Server specific type: https://learn.microsoft.com/dotnet/api/system.data.sqldbtype?view=net-6.0

>Specifies SQL Server-specific data type of a field, property, for use in a [SqlParameter](https://learn.microsoft.com/en-us/dotnet/api/system.data.sqlclient.sqlparameter?view=net-6.0).

I don't want to use naming that conflicts with pre-existing naming scheme for enums provided by dotnet/ relevant ADO.NET Data Provider. 

PgSql/MySql will not use `SqlDbType`. For example, PG (Npgsql) uses the following table for data type mappings which include usage of the enum `NpgsqlDbType ` -> `DbType` https://www.npgsql.org/doc/types/basic.html

Reply:
I see. Makes sense!